### PR TITLE
feat(VList): port to v3

### LINKED ISSUE #8137: [Bug Report] v-subheaders no longer automatically collapse in a mini nav-menu
### Environment
**Vuetify Version:** 2.0.1
**Last working version:** 1.5.16
**Vue Version:** 2.6.10
**Browsers:** Chrome 75.0.3770.142
**OS:** Windows 10

### Steps to reproduce
In the codepen open the nav drawer via the toggle button. Then toggle to mini mode, via another toggle button. You'll notice the v-subheader is still in the mini nav menu and the text looks terrible.

### Expected Behavior
via css classes v-subheaders should be hidden when in a nav menu with class "mini"

### Actual Behavior
the v-subheaders remain visible in the mini nav menu

### Reproduction Link
<a href="https://codepen.io/anon/pen/bXgYxb?&page=1" target="_blank">https://codepen.io/anon/pen/bXgYxb?&page=1</a>

### LINKED ISSUE #9285: [Bug Report] Impossible to set color of List Item text
### Environment
**Vuetify Version:** 2.1.1
**Vue Version:** 2.6.10
**Browsers:** Chrome 77.0.3865.90
**OS:** Mac OS 10.14.6

### Steps to reproduce
Try to set color (text/icon) of not active list item with available tools:
- `color` prop on list item;
- `active-class` prop on list item;
- `class` on list item;
- css;
- css with `!important`;
- inline css;

### Expected Behavior
image

### Actual Behavior
image

Impossible to replace light* text color of dark list item without inline css.
With inline css active tile is effected and cannot be styled to have dark* text color.

*light = the white color used for light theme backgrounds and dark theme texts.
*dark = the black color used for dark theme backgrounds and white theme texts.

image

### Reproduction Link
<a href="https://codepen.io/JesusCrow/pen/OJJPqVb?editors=0100" target="_blank">https://codepen.io/JesusCrow/pen/OJJPqVb?editors=0100</a>

### Other comments
I tried everything I could think up to create a list item that has a certain color for text and icon. I think that this is impossible without inline css. Using inline css disables me from styling active item text.

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Specification
* <https://material.io/components/lists>

## Motivation and Context

* fixes #8137
* fixes #9285

## Markup:

<details>

```vue
https://gist.github.com/johnleider/adc9f00de0674b95ee4a92f19fbca028
```

</details>

#### Bug fixes

* n/a

#### Features

**Components**

* added `v-list-img`
* added `v-list-subheader`

**Styles**

* added $list-background
* added $list-border-color
* added $list-border-style
* added $list-border-thin-width
* added $list-border-width
* added $list-color
* added $list-density
* added $list-disabled-opacity
* added $list-positions
* added $list-subheader-font-size
* added $list-subheader-font-weight
* added $list-subheader-inset-margin
* added $list-subheader-inset-padding-start
* added $list-subheader-line-height
* added $list-subheader-min-height-multiplier
* added $list-subheader-min-height
* added $list-subheader-padding-end
* added $list-subheader-padding-start
* added $list-subheader-text-opacity
* added $list-subheader-transition

#### Breaking changes

**Props**

* removed **dense** prop, use **density="compact"**
* removed **expand** prop, use **density="compact"**
* removed **flat** prop, elevation is 0 by default
* removed **two-line** prop, use **lines="two"**
* removed **three-line** prop, use **lines="three"**

**Styles**

* removed $avatar-margin-x
* removed $list-dense-content-padding
* removed $list-dense-icon-height
* removed $list-dense-icon-margin
* removed $list-dense-min-height
* removed $list-dense-subheader-font-size
* removed $list-dense-subheader-height
* removed $list-dense-subheader-padding
* removed $list-dense-subheading-padding
* removed $list-nav-border-radius
* removed $list-nav-item-padding
* removed $list-nav-padding-left
* removed $list-nav-padding-right
* removed $list-nav-rounded-dense-item-margin-bottom
* removed $list-nav-rounded-item-dense-margin-bottom
* removed $list-nav-rounded-item-margin-bottom
* removed $list-shaped-border-radius
* removed $list-shaped-padding

#### Bug fixes

* n/a

#### Features

**Components**

* added `v-list-item-header`
* added `v-list-item-media`
* added `v-list-item-img`

**Props**

* added **active** prop, replaces existing **value** prop
* added **active-color** prop, applies color when component is active
* added **contained** prop, changes how color is applied to the component
* added **append-avatar** prop
* added **prepend-avatar** prop
* added **append-icon** prop
* added **prepend-icon** prop
* added **title** prop - generate and apply prop to the `v-list-item-title` component
* added **subtitle** prop - generate and apply prop to the `v-list-item-subtitle` component
* added **left** && **right** prop to `v-list-item-avatar`

**Styles**

* added $list-item-avatar-align-self
* added $list-item-avatar-margin-end
* added $list-item-avatar-margin-start
* added $list-item-avatar-margin-y
* added $list-item-avatar-size
* added $list-item-avatar-size-multiplier
* added $list-item-media-margin-bottom
* added $list-item-media-margin-end
* added $list-item-media-margin-start
* added $list-item-media-margin-top
* added $list-item-media-three-line-margin-bottom
* added $list-item-media-three-line-margin-top
* added $list-item-media-two-line-margin-bottom
* added $list-item-media-two-line-margin-top
* added $list-item-nav-subtitle-font-size
* added $list-item-nav-subtitle-font-weight
* added $list-item-nav-subtitle-letter-spacing
* added $list-item-nav-subtitle-line-height
* added $list-item-nav-title-font-size
* added $list-item-nav-title-font-weight
* added $list-item-nav-title-letter-spacing
* added $list-item-nav-title-line-height
* added $list-item-padding
* added $list-item-subtitle-font-weight
* added $list-item-subtitle-letter-spacing
* added $list-item-subtitle-line-height
* added $list-item-subtitle-opacity
* added $list-item-subtitle-padding
* added $list-item-subtitle-text-transform
* added $list-item-three-line-padding
* added $list-item-title-font-size
* added $list-item-title-font-weight
* added $list-item-title-header-padding
* added $list-item-title-hyphens
* added $list-item-title-letter-spacing
* added $list-item-title-line-height
* added $list-item-title-overflow-wrap
* added $list-item-title-padding
* added $list-item-title-text-overflow
* added $list-item-title-text-transform
* added $list-item-title-word-break
* added $list-item-title-word-wrap
* added $list-item-transition
* added $list-item-two-line-padding

#### Breaking Changes

**Components**

* removed `v-list-item-action-text`
* removed `v-list-item-action`
* removed `v-list-item-content`, use `v-list-item-header`
* removed `v-list-item-icon`, use `v-list-item-avatar icon="..." />`
* removed **size** and **horizontal** prop from `v-list-item-avatar`

**Props**

* removed **two-line** prop, wrap with `v-list lines="two"`
* removed **three-line** prop, wrap with `v-list lines="three"`
* removed **selectable** prop
* removed **dense** prop, use **density="compact"**
* removed **inactive** prop
* removed **value** prop, use **active**

**Styles**

* removed $list-item-action-icon-margin
* removed $list-item-action-margin
* removed $list-item-action-text-font-size
* removed $list-item-avatar-first-child-margin
* removed $list-item-avatar-horizontal-margin
* removed $list-item-avatar-horizontal-margin-x
* removed $list-item-avatar-margin-y
* removed $list-item-child-last-type-margin
* removed $list-item-child-min-width
* removed $list-item-content-children-margin-bottom
* removed $list-item-content-padding
* removed $list-item-dense-three-line-min-height
* removed $list-item-dense-title-font-size
* removed $list-item-dense-title-line-height
* removed $list-item-dense-two-line-min-height
* removed $list-item-icon-margin
* removed $list-item-icon-margin-y
* removed $list-item-padding
* removed $list-item-subtitle-font-size!default;
* removed $list-item-three-line-avatar-action-margin

### VListGroup

**Breaking Changes**

* removed component, being re-implemented in a future patch to `v-list`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VList/VListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    const { elevationClasses } = useElevation(props)
+    const { positionClasses, positionStyles } = usePosition(props, 'v-list-item')
+    const { roundedClasses } = useRounded(props, 'v-list-item')
+    const { densityClasses } = useDensity(props, 'v-list-item')
+
+    return () => {
+      const hasTitle = (slots.title || props.title)
+      const hasSubtitle = (slots.subtitle || props.subtitle)
+      const hasHeader = (hasTitle || hasSubtitle)
+      const hasAppend = (slots.append || props.appendAvatar || props.appendIcon)
```
Comment (nekosaur) [inline packages/vuetify/src/components/VList/VListItem.tsx:67]: I thought we said we'd only have props for icons, otherwise use slot?

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/core-team

File: packages/vuetify/src/components/VList/_variables.scss [hunk 2]
```diff
@@ -1,67 +1,78 @@
...
+$list-item-subtitle-padding: 0 !default;
+$list-item-subtitle-text-transform: none !default;
+
+$list-item-title-font-size: map-deep-get($typography, 'body-1', 'size') !default;
+$list-item-title-font-weight: map-deep-get($typography, 'body-1', 'weight') !default;
+$list-item-title-header-padding: 0 !default;
+$list-item-title-hyphens: auto !default;
+$list-item-title-letter-spacing: map-deep-get($typography, 'body-1', 'letter-spacing') !default;
+$list-item-title-line-height: map-deep-get($typography, 'body-1', 'line-height') !default;
+$list-item-title-overflow-wrap: anywhere !default;
```
Comment (KaelWD) [inline packages/vuetify/src/components/VList/_variables.scss:46]: Not supported by Safari

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/core-team

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/core-team

**Pull request merged**